addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/common/HomematicConfig.java

```diff
@@ -0,0 +1,263 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.homematic.internal.common;
+
+import org.apache.commons.lang.builder.ToStringBuilder;
+import org.apache.commons.lang.builder.ToStringStyle;
+import org.openhab.binding.homematic.internal.model.HmChannel;
+import org.openhab.binding.homematic.internal.model.HmGatewayInfo;
+import org.openhab.binding.homematic.internal.model.HmInterface;
+
+/**
+ * The main gateway config class.
+ *
+ * @author Gerhard Riegler - Initial contribution
+ */
+public class HomematicConfig {
+    private static final String ISO_ENCODING = "ISO-8859-1";
+    private static final String UTF_ENCODING = "UTF-8";
+
+    public static final String GATEWAY_TYPE_AUTO = "AUTO";
+    public static final String GATEWAY_TYPE_CCU = "CCU";
+
+    private static final int DEFAULT_PORT_RF = 2001;
+    private static final int DEFAULT_PORT_WIRED = 2000;
+    private static final int DEFAULT_PORT_CUXD = 8701;
+
+    private String gatewayAddress;
+    private String gatewayType = GATEWAY_TYPE_AUTO;
+
+    private int rfPort;
```

Comment:
Note: Valid field types are only String, BigDecimal and Boolean, see https://www.eclipse.org/smarthome/documentation/development/bindings/how-to.html#configuration